backend/pkg/broker/topics/blcu/download.go

```diff
@@ -1,29 +1,75 @@
 package blcu
 
 import (
+	"encoding/json"
+	"fmt"
 	"github.com/HyperloopUPV-H8/h9-backend/pkg/abstraction"
+	"github.com/HyperloopUPV-H8/h9-backend/pkg/boards"
 	"github.com/HyperloopUPV-H8/h9-backend/pkg/websocket"
 )
 
 const DownloadName abstraction.BrokerTopic = "blcu/download"
 
 type Download struct {
-	pool *websocket.Pool
-	api  abstraction.BrokerAPI
+	pool   *websocket.Pool
+	api    abstraction.BrokerAPI
+	client websocket.ClientId
 }
 
 func (download *Download) Topic() abstraction.BrokerTopic {
 	return DownloadName
 }
 
 func (download *Download) Push(push abstraction.BrokerPush) error {
+	switch push.Topic() {
+	case boards.DownloadSuccess{}.Topic():
+		err := download.pool.Write(download.client, websocket.Message{
+			Topic:   push.Topic(),
+			Payload: nil,
+		})
+		if err != nil {
+			return err
+		}
+	case boards.DownloadFailure{}.Topic():
+		err := download.pool.Write(download.client, websocket.Message{
+			Topic:   push.Topic(),
+			Payload: nil,
+		})
+		if err != nil {
+			return err
+		}
+	}
+
 	return nil
 }
 
 func (download *Download) Pull(request abstraction.BrokerRequest) (abstraction.BrokerResponse, error) {
 	return nil, nil
 }
 
+func (download *Download) ClientMessage(id websocket.ClientId, message *websocket.Message) {
+	download.client = id
+
+	switch message.Topic {
+	case DownloadName:
+		err := download.handleDownload(message)
+		if err != nil {
+			fmt.Printf("error handling download: %v\n", err)
+		}
+	}
+}
+
+func (download *Download) handleDownload(message *websocket.Message) error {
+	var downloadEvent boards.DownloadEvent
```

Comment:
add an struct in between the JSON and the boards.DownloadEvent. Unmarshal the payload to that JSON and then take that JSON fields and use it to construct the DownloadEvent.
See how it worked last year to make sure the JSON structure matches

Reply:
It seems that last year we used:
`type downloadRequest struct {
	Board string `json:"board"`
}`